### PR TITLE
Add `php82` from PHP 8.2.0alpha2 release.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: pkgs/php/8.2.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{ prev, mkPhp, php82 ? null}:
+
+let
+  base = mkPhp {
+    version = "8.2.0alpha2";
+    sha256 = null;
+
+    phpAttrsOverrides = attrs: {
+      src = php82;
```
Comment (jtojnar) [inline pkgs/php/8.2.nix:9]: I do not see much benefit in specifying it in `flake.nix` when it is not easily updateable, just use `fetchurl` here.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -64,6 +65,7 @@ jobs:
         run: nix-build -A outputs.checks.x86_64-linux.${{ steps.params.outputs.attr }}-mysql
 
       - name: Build Xdebug extension
+        if: ${{ steps.params.outputs.major < 8 && steps.params.outputs.minor < 2 }}
```
Comment (jtojnar) [inline .github/workflows/build.yaml:68]: Please also skip it in `checks.nix` so that `nix flake check` can pass.

File: pkgs/php/8.2.nix (hunk 1 above)
Comment (drupol) [reply pkgs/php/8.2.nix]: Right, fixed !

File: .github/workflows/build.yaml (hunk 2 above)
Comment (drupol) [reply .github/workflows/build.yaml:68]: Done, I hope it's ok.

File: checks.nix [hunk 3]
```diff
@@ -37,7 +37,7 @@ let
 
     xdebug = {
       description = "Build Xdebug extension";
-      enabled = { php, lib, ... }: lib.versionOlder php.version "81";
+      enabled = { php, lib, ... }: php.version != "8.2.0alpha2";
```
Comment (jtojnar) [inline checks.nix:40]: Try `lib.versionOlder php.version "8.2";`

File: pkgs/phps.nix [hunk 4]
```diff
@@ -1,4 +1,6 @@
-nixpkgs:
+{
```
Comment (jtojnar) [inline pkgs/phps.nix:1]: These changes probably do not belong into this commit.

Comment (drupol) [reply pkgs/phps.nix]: Fixed!

Comment (drupol) [reply pkgs/phps.nix:1]: When removing that, I need to change :

```nix
      overlays.default = import ./pkgs/phps.nix {
        inherit nixpkgs;
      };
```

in

```nix
      overlays.default = import ./pkgs/phps.nix nixpkgs.outPath;
```

Should I do it?

Comment (drupol) [reply pkgs/phps.nix:1]: Honestly, I prefer leaving it as such, WDYT ?

Comment (jtojnar) [reply pkgs/phps.nix:1]: Yeah, I would just keep it as it was before. The refactoring is okay but it is not really related to this change so I would prefer to keep it separated.

File: flake.nix [hunk 5]
```diff
@@ -10,10 +10,15 @@
 
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
 
+    php82 = {
+      url = "https://downloads.php.net/~pierrick/php-8.2.0alpha2.tar.xz";
+      flake = false;
+    };
+
```
Comment (jtojnar) [inline flake.nix]: Could you also squash the [Remove flake input and use fetchurl.](https://github.com/fossar/nix-phps/pull/118/commits/9c3a2cce89f8f471b461fec6b923f5f99b114c9a) into the parent commit? I do not think we need this in history.

File: pkgs/phps.nix (hunk 4 above)
Comment (drupol) [reply pkgs/phps.nix:1]: Okidocky.

File: flake.nix (hunk 5 above)
Comment (drupol) [reply flake.nix]: Yep, will do it.

File: pkgs/php/8.2.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      exif
+      fileinfo
+      filter
+      ftp
+      gd
+      gettext
+      gmp
+      iconv
+      intl
+      # json
```
Comment (jtojnar) [inline pkgs/php/8.2.nix:35]: Why json here?

File: pkgs/phps.nix [hunk 7]
```diff
@@ -90,8 +90,7 @@ let
                 );
             }
           );
-    }
-    // args;
+    } // args;
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/phps.nix]: fixed

**Pull request merged**